sympy/matrices/expressions/matexpr.py

```diff
@@ -1269,6 +1270,46 @@ def _make_matrix(x):
     return ImmutableDenseMatrix([[x]])
 
 
+class MatrixSet(Set):
+    """
+    MatrixSet represents the set of matrices with ``shape = (n, m)`` over the
+    given set.
+
+    Examples
+    ========
+
+    >>> from sympy.matrices import MatrixSet, Matrix
+    >>> from sympy import S, I
+    >>> M = MatrixSet(2, 2, set=S.Reals)
+    >>> X = Matrix([[1, 2], [3, 4]])
+    >>> X in M
+    True
+    >>> X = Matrix([[1, 2], [I, 4]])
+    >>> X in M
+    False
+
+    """
+    is_iterable = False
+    is_empty = False
+
+    def __new__(cls, n, m, set):
+        n, m, set = _sympify(n), _sympify(m), _sympify(set)
+        return Set.__new__(cls, n, m, set)
+
+    is_finite_set = property(lambda self: self.set.is_finite_set)
+    is_ComplexRegion = property(lambda self: self.set.is_ComplexRegion)
+
+    shape = property(lambda self: self.args[0:2])
+    set = property(lambda self: self.args[2])
+
+    def _contains(self, other):
+        if not isinstance(other, MatrixExpr):
+            raise TypeError("%s should be an instance of MatrixExpr" % str(other))
+        if other.shape != self.shape:
```

Comment:
This should return None if one have symbolic shape and the other have numeric shape, or much more cases when the equality of shape is logically undetermined.

Reply:
Is there something similar in SymPy, we should be consistent for outputs.  Otherwise, agreed with @sylee957